addons/project_timesheet_holidays/tests/test_timesheet_global_time_off.py

```diff
@@ -175,3 +188,92 @@ def test_timesheet_creation_and_deletion_for_calendar_set_and_remove(self):
+        # So it should add to 2(full time employees)*5(leave days)*8(hours per day) + 2(part time employees)*4(leave days)*6(hours per day).
+        self.assertEqual(leave_task.effective_hours, 128)
+
+
+        # Now we set the calendar_id. The timesheets should be deleted from other calendars.
+        global_time_off.calendar_id = self.test_company.resource_calendar_id.id
+        timesheets_by_employee = self._get_timesheets_by_employee(leave_task)
+
+        self.assertFalse(timesheets_by_employee.get(self.part_time_employee), False)
+        self.assertFalse(timesheets_by_employee.get(self.part_time_employee2), False)
+        self.assertEqual(len(timesheets_by_employee.get(self.full_time_employee)), 5)
+        self.assertEqual(len(timesheets_by_employee.get(self.full_time_employee_2)), 5)
+        self.assertEqual(leave_task.effective_hours, 80)
+
+    def test_timesheet_creation_for_global_time_off_in_differant_company(self):
+        leave_start_datetime = datetime(2021, 1, 4, 7, 0, 0, 0)  # This is a monday
+        leave_end_datetime = datetime(2021, 1, 8, 18, 0, 0, 0)  # This is a friday
+
+        new_company = self.env['res.company'].create({
+            'name': 'Winterfell',
+        })
+
+        self.env['resource.calendar.leaves'].with_company(new_company).create({
+            'name': 'Test',
+            'calendar_id': False,
+            'date_from': leave_start_datetime,
+            'date_to': leave_end_datetime,
+        })
+
+        leave_task = self.test_company.leave_timesheet_task_id
+        timesheets_by_employee = self._get_timesheets_by_employee(leave_task)
+        # Should no create timesheet if leave is in differant company then employees
+        self.assertFalse(timesheets_by_employee.get(self.part_time_employee, False))
+        self.assertFalse(timesheets_by_employee.get(self.full_time_employee, False))
+        # Should not add any timsheets in other companies
+        self.assertEqual(leave_task.effective_hours, 0)
+
+    def test_timesheet_creation_for_global_time_off_wo_calendar_in_batch(self):
+
+        self.env['resource.calendar.leaves'].with_company(self.test_company).create([{
+            'name': "Easter Monday",
+            'calendar_id': False,
+            'date_from': datetime(2022, 4, 18, 5, 0, 0),
+            'date_to': datetime(2022, 4, 18, 18, 0, 0),
+            'resource_id': False,
+            'time_type': "leave",
+        }, {
+            'name': "Ascension Day",
+            'calendar_id': False,
+            'date_from': datetime(2022, 4, 26, 5, 0, 0),
+            'date_to': datetime(2022, 4, 26, 18, 0, 0),
+        }])
+
+        # 2 Timesheets for 2 global leaves should have been created for current companies all calendar employees
+        leave_task = self.test_company.leave_timesheet_task_id
+        timesheets_by_employee = self._get_timesheets_by_employee(leave_task)
+
+        self.assertEqual(len(timesheets_by_employee.get(self.part_time_employee)), 2)
```

Comment:
```suggestion
        self.assertEqual(len(timesheets_by_employee.get(self.part_time_employee, [])), 2)
```
To have a prettier error, maybe?

Reply:
No need after applying read_group